telethon/tl/tlobject.py

```diff
@@ -121,21 +135,21 @@ def serialize_bytes(data):
 
     @staticmethod
     def serialize_datetime(dt):
-        if not dt:
+        if not dt and not isinstance(dt, timedelta):
```

Comment:
So now `datetime`, `date`, `float` and `int` won't work?

Reply:
They work, it returns `b'\0\0\0\0'` only if it is not a timedelta and `bool(dt) == 0` (which could be true for float, int, None, False, etc). For datetime / date (where bool(dt) is != 0 anyways) / non-0 float/int, applies the rest of the `.serialize_datetime`, like before.

Reply:
Oh good point, brain fart, my bad.